### DISCUSSION TITLE
AirPlay alternative / Phone/Tablet -> Google Cast  -> Snapserver ?

### DISCUSSION THREAD

**Discussion by jbrechtel** [snapcast/snapcast#1264] [Q&A]
Hi,

I'm curious if anyone knows of an open source project that can masquerade as a Chromecast / Google Cast device and then feed its received audio to STDOUT or a pipe for consumption by Snapcast?

Generally speaking, the experience I'm after is to be able to take something playing on my tablet or phone and cast its audio to a "Snapcast" receiver via the "Cast" button - but honestly if there were another way that would be fine, too.

It seems like the Chromecast part is entirely proprietary - so maybe it's a dead-end to look for a Chromecast option for this.

I know Shairport-sync and AirMusic exist and that works but it's a little kludgy compared to the Chromecast experience.

Are there any other options for this?

**Comment by DerPicknicker** [comment]
From which device you want to stream? 

I use now music assistant and it works brilliantly. In later versions they add support for emulating a Spotify connect and an AirPlay device.

**Reply by igbjnI05bF** [reply]
I made a similar discussion post the other day before seeing this - I'd love the same thing, the ability to send audio from an android (grapheneOS) phone to snapcast. In the same way that you can connect a bluetooth device to an android phone and send *all* audio to it, I'd like to do that with snapcast somehow. I'd love the ability to listen to a podcast using any app, or even through a mobile web browser, and have the audio play to snapcast. Even answer a phone call if I wanted and hear the output through snapcast.

This is very possible using pipewire on linux to discover and choose a local snapcast server as the output device, just want the same thing for android.

I do have music assistant installed and it's nice to be able to play my jellyfin collection to snapcast, but that still doesn't seem to allow me to send audio from my android to it unless I'm missing something.

**Reply by DerPicknicker** [reply]
@igbjnI05bF .. sorry for my late reply... 

If you have root access you can later use AirPlay for system audio.. 

If you use Spotify it will be available also as player. 

So it highly depends on your requirements;-)

**Reply by igbjnI05bF** [reply]
I have GrapheneOS which to my knowledge doesn't allow root access to the phone. I know that Spotify will let me send to Snapcast via librespot but I'm looking for the ability to send *any/all* audio from my android phone to snapcast. I'd like to be able to be able to open the app AudioBookShelf and send that audio to snapcast, or even open up firefox in a web browser and have audio from firefox be sent to snapcast.

The only comparison I can make is to do it the same way that bluetooth does it. When you connect bluetooth earbuds to an android phone, *all* audio is sent to the earbuds. I want to do that, but send it to the snapcast server.

**Reply by LeoSum8** [reply]
You could take a look at cornrow.
https://github.com/badaix/snapcast/discussions/911

**Reply by jbrechtel** [reply]
Apologies for abandoning this thread!

@DerPicknicker - Generally speaking I want to be able to click the "cast" button on YouTube or other apps which support it on Android and have the audio from that stream play via Snapcast.

I have used AirPlay successfully in the past and it's ... mostly fine ... but not perfect since generally the phone itself is still responsible for receiving the audio and then sending it back over the WiFi network to Snapcast. That results in not-the-best UX and some lag.

That's why I was looking for Chromecast support - just like this poor soul was - https://github.com/badaix/snapcast/discussions/1403
